acceptancetests/deploy_stack.py

```diff
@@ -736,6 +771,32 @@ def from_args(cls, args):
         return cls.from_client(args, client)
 
     @classmethod
+    def from_existing(cls, args):
+        try:
+            juju_home = os.environ['JUJU_HOME']
```

Comment:
This should really be JUJU_DATA as JUJU_HOME is a juju1 naming scheme.
You could cover all bases with: os.environ.get('JUJU_DATA', os.environ['JUJU_HOME'])
which will error if neither JUJU_DATA or JUJU_HOME are set. Then have the error message state JUJU_DATA.

Although I know this is at odds with what jujupy currently expects, it's a point of confusion for people using the tests and framework. We should (probably as a separate PR :-)) update jujupy to expect JUJU_DATA (maybe as well as JUJU_HOME) and move away from using JUJU_HOME as it's really outdated now and juju1 runs are the exception.